plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsWriter.java

```diff
@@ -194,51 +189,58 @@ private CollectedStatistics mergeStatisticsIfNecessary(
                 updateMode = REPLACE;
             }
         }
+        Map<Integer, CompactSketch> collectedNdvSketches = collectedStatistics.ndvSketches();
+        StatsUpdateMode finalUpdateMode = updateMode;
+        Set<Integer> columnsWithRecentlyComputedStats = new HashSet<>(collectedNdvSketches.keySet());
 
-        return switch (updateMode) {
-            case REPLACE -> collectedStatistics;
-            case INCREMENTAL_UPDATE -> {
```

Comment:
I am looking at this diff and i don't understand why this is being removed.
Wasn't the previous structure OK?

same for `mergeStatisticsIfNecessary` return type. What was wrong about it?
if, in the future, we collect anything else besides NDV (`CollectedStatistics` changes), we want the `mergeStatisticsIfNecessary` to be updated appropriately, so having the method return `CollectedStatistics` looked right to me